<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Single cell data integrative analysis â€¢ scdney</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Single cell data integrative analysis">
<meta property="og:description" content="A collection of single cell RNA-sequencing data analysis functions including normalisation, mixture modelling, clustering and cell type correction.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scdney</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/scdney.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">

<div id="scdney---single-cell-data-integrative-analysis" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#scdney---single-cell-data-integrative-analysis" class="anchor"></a>scdney - Single cell data integrative analysis</h1></div>
<p><code>scdney</code> is a R package with collection of single cell RNA-sequencing (scRNA-seq) data analysis functions developed by team of <a href="http://www.maths.usyd.edu.au/u/SMS/bioinformatics/"><strong>Sydney Precision Bioinformatics Research Group</strong></a> at The University of Sydney.</p>
<p>This package contains useful functions for analysis of scRNA-seq data including clustering, cell type identification, etc.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"SydneyBioX/scdney"</span>, <span class="dt">build_opts =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"--no-resave-data"</span>, <span class="st">"--no-manual"</span>))
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scdney)</code></pre></div>
<p>For <code>devtools (&lt; 2.0.0)</code>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"SydneyBioX/scdney"</span>, <span class="dt">build_vignettes =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scdney)</code></pre></div>
<p>Building the vignette may take some time. If you wish not to create the vignette during installation, try:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"SydneyBioX/scdney"</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scdney)</code></pre></div>
<p><strong>NOTE:</strong> <em>For mac users, the official cran mirror of R tools for OS X and R tools for OS X on r.research.att.com that lists the gfortran binary are out of date. You will need to update <code>gfortran</code> and add the following line <code>FLIBS=-L/usr/local/Cellar/gcc/X.Y.Z/lib/gcc/X</code> (where <code>X.Y.Z</code> is your gcc version) to <code>~/.R/Makevars</code> prior to this package installation.</em></p>
</div>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<div id="section-1---clustering" class="section level2">
<h2 class="hasAnchor">
<a href="#section-1---clustering" class="anchor"></a>Section 1 - Clustering</h2>
<div id="scclust" class="section level3">
<h3 class="hasAnchor">
<a href="#scclust" class="anchor"></a>scClust</h3>
<p><img src="./img/scClust.jpg" alt="alt text"></p>
<p>Current version of this package is implemented to run SIMLR (Wang et al, 2017) or k-means clustering methods with various similarity metrics.</p>
<p>Available metrics include:</p>
<p>SIMLR - <code>"pearson"</code> correlation, <code>"spearman"</code> corelation and <code>"euclidean"</code> distance.</p>
<p>K-means - <code>"pearson"</code> correlation, <code>"spearman"</code> correlation, <code>"euclidean"</code> distance, <code>"manhattan"</code> distance and <code>"maximum"</code> distance.</p>
<div id="load-data" class="section level4">
<h4 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load Data</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(GSE82187.sample)
mat &lt;-<span class="st"> </span>GSE82187

mat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log2</a></span>(mat<span class="op">+</span><span class="dv">1</span>)

<span class="co"># set number of clusters (classes defined in colnames)</span>
nCs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(mat))</code></pre></div>
</div>
<div id="part-a--clustering-with-different-similarity-metrics-with-scclust" class="section level4">
<h4 class="hasAnchor">
<a href="#part-a--clustering-with-different-similarity-metrics-with-scclust" class="anchor"></a>Part A. Clustering with different similarity metrics with <code>scClust</code>
</h4>
<p>To run scClust,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SIMLR</span>
simlr.result &lt;-<span class="st"> </span><span class="kw"><a href="reference/scClust.html">scClust</a></span>(mat, nCs, <span class="dt">similarity =</span> <span class="st">"pearson"</span>, <span class="dt">method =</span> <span class="st">"simlr"</span>, <span class="dt">seed =</span> <span class="dv">1</span>, <span class="dt">cores.ratio =</span> <span class="dv">0</span>)

<span class="co"># K-means</span>
kmeans.result &lt;-<span class="st"> </span><span class="kw"><a href="reference/scClust.html">scClust</a></span>(mat, nCs, <span class="dt">similarity =</span> <span class="st">"pearson"</span>, <span class="dt">method =</span> <span class="st">"kmeans"</span>, <span class="dt">seed =</span> <span class="dv">1</span>, <span class="dt">nstart =</span> <span class="dv">10</span>, <span class="dt">iter.max =</span> <span class="dv">10</span>)</code></pre></div>
<p>This function allows you to perform clustering with a user specified similarity metrics. The return values of <code>scClust</code> are identical to clustering methods for <code>Kmeans</code> and <code>SIMLR</code> functions.</p>
</div>
<div id="part-b--benchmarking-different-similarity-metrics-with-scclustbench" class="section level4">
<h4 class="hasAnchor">
<a href="#part-b--benchmarking-different-similarity-metrics-with-scclustbench" class="anchor"></a>Part B. Benchmarking different similarity metrics with <code>scClustBench</code>
</h4>
<p>This section is to compare a set of similarity metrics on clustering methods to benchmark their perfomance accuracy.</p>
<p>To run scClustBench,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># SIMLR</span>
simlr.result &lt;-<span class="st"> </span><span class="kw"><a href="reference/scClustBench.html">scClustBench</a></span>(mat, nCs, <span class="dt">method =</span> <span class="st">"simlr"</span>, <span class="dt">rep =</span> <span class="dv">2</span>, <span class="dt">seed =</span> <span class="dv">1</span>, <span class="dt">cores =</span> <span class="dv">1</span>, <span class="dt">cores.ratio =</span> <span class="dv">0</span>)

<span class="co"># K-means</span>
kmeans.result &lt;-<span class="st"> </span><span class="kw"><a href="reference/scClustBench.html">scClustBench</a></span>(mat, nCs, <span class="dt">method =</span> <span class="st">"kmeans"</span>, <span class="dt">rep =</span> <span class="dv">2</span>, <span class="dt">seed =</span> <span class="dv">1</span>, <span class="dt">cores =</span> <span class="dv">1</span>, <span class="dt">nstart =</span> <span class="dv">10</span>, <span class="dt">iter.max =</span> <span class="dv">10</span>)</code></pre></div>
<p>You can evaluate this result with the function <code>evalScClustBench</code> and plot with <code>plotSimlrEval</code> or <code>plotKmeansEval</code>.</p>
<p>For further demonstrations, see:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/browseVignettes">browseVignettes</a></span>(<span class="st">"scdney"</span>)</code></pre></div>
</div>
</div>
</div>
<div id="section-2---post-hoc-cell-type-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#section-2---post-hoc-cell-type-classification" class="anchor"></a>Section 2 - Post hoc cell type classification</h2>
<div id="screclassify" class="section level3">
<h3 class="hasAnchor">
<a href="#screclassify" class="anchor"></a>scReClassify</h3>
<p><img src="./img/scReClassify.jpg" alt="alt text"></p>
<p>Current version of this package is implemented to run with <code>svm</code> and <code>radomForest</code> classifiers.</p>
<div id="load-data-1" class="section level4">
<h4 class="hasAnchor">
<a href="#load-data-1" class="anchor"></a>Load data</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(GSE87795_liver.development.data)
dat &lt;-<span class="st"> </span>GSE87795_liver.development.data<span class="op">$</span>data
cellTypes &lt;-<span class="st"> </span>GSE87795_liver.development.data<span class="op">$</span>cellTypes

<span class="co"># number of clusters</span>
nCs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cellTypes))

<span class="co"># This demo dataset is already pre-processed</span>
dat.processed =<span class="st"> </span>dat</code></pre></div>
</div>
<div id="part-a--screclassify-demonstration" class="section level4">
<h4 class="hasAnchor">
<a href="#part-a--screclassify-demonstration" class="anchor"></a>Part A. scReClassify (Demonstration)</h4>
<div id="dimension-reduction" class="section level5">
<h5 class="hasAnchor">
<a href="#dimension-reduction" class="anchor"></a>Dimension reduction</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat.selected =<span class="st"> </span><span class="kw"><a href="reference/matPCs.html">matPCs</a></span>(dat.processed, <span class="fl">0.7</span>)</code></pre></div>
</div>
<div id="synthetic-noise-demonstration-purpose" class="section level5">
<h5 class="hasAnchor">
<a href="#synthetic-noise-demonstration-purpose" class="anchor"></a>Synthetic noise (Demonstration purpose)</h5>
<p>Here in this example, we will synthetically generate varying degree of noise in sample labels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lab &lt;-<span class="st"> </span>cellTypes

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">1</span>)
noisyCls &lt;-<span class="st"> </span><span class="cf">function</span>(dat, rho, cls.truth){
  cls.noisy &lt;-<span class="st"> </span>cls.truth
  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(cls.noisy) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(dat)
  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))) {
    <span class="co"># class label starts from 0</span>
    <span class="cf">if</span> (i <span class="op">!=</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))) {
      cls.noisy[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cls.truth <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[i]), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Round">floor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(cls.truth <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[i]) <span class="op">*</span><span class="st"> </span>rho))] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[i<span class="op">+</span><span class="dv">1</span>]
    } <span class="cf">else</span> {
      cls.noisy[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cls.truth <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[i]), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Round">floor</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(cls.truth <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[i]) <span class="op">*</span><span class="st"> </span>rho))] &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cls.noisy))[<span class="dv">1</span>]
    }
  }

  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(cls.truth <span class="op">!=</span><span class="st"> </span>cls.noisy))
  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/function">return</a></span>(cls.noisy)
}

cls.noisy01 &lt;-<span class="st"> </span><span class="kw">noisyCls</span>(dat.selected, <span class="dt">rho=</span><span class="fl">0.1</span>, lab)
cls.noisy02 &lt;-<span class="st"> </span><span class="kw">noisyCls</span>(dat.selected, <span class="dt">rho=</span><span class="fl">0.2</span>, lab)
cls.noisy03 &lt;-<span class="st"> </span><span class="kw">noisyCls</span>(dat.selected, <span class="dt">rho=</span><span class="fl">0.3</span>, lab)
cls.noisy04 &lt;-<span class="st"> </span><span class="kw">noisyCls</span>(dat.selected, <span class="dt">rho=</span><span class="fl">0.4</span>, lab)
cls.noisy05 &lt;-<span class="st"> </span><span class="kw">noisyCls</span>(dat.selected, <span class="dt">rho=</span><span class="fl">0.5</span>, lab)</code></pre></div>
</div>
<div id="use-screclassify-with-adasampling-to-correct-mislabelled-cell-types-" class="section level5">
<h5 class="hasAnchor">
<a href="#use-screclassify-with-adasampling-to-correct-mislabelled-cell-types-" class="anchor"></a>Use scReClassify with AdaSampling to correct mislabelled cell types.</h5>
<p>Here, we will only Support Vector machine (<code>svm</code>) as base classifier.</p>
</div>
</div>
<div id="benchmark-evaluation" class="section level4">
<h4 class="hasAnchor">
<a href="#benchmark-evaluation" class="anchor"></a>Benchmark Evaluation</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">###################################
<span class="co"># SVM</span>
###################################
acc01 &lt;-<span class="st"> </span>acc02 &lt;-<span class="st"> </span>acc03 &lt;-<span class="st"> </span>acc04 &lt;-<span class="st"> </span>acc05 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>()
ari01 &lt;-<span class="st"> </span>ari02 &lt;-<span class="st"> </span>ari03 &lt;-<span class="st"> </span>ari04 &lt;-<span class="st"> </span>ari05 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>()
base &lt;-<span class="st"> "svm"</span>

<span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>) {
  final &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.selected, cls.noisy01, <span class="dt">seed=</span>j, <span class="dt">classifier=</span>base, <span class="dt">percent=</span><span class="dv">1</span>, <span class="dt">L=</span><span class="dv">10</span>)<span class="op">$</span>final
  ari01 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ari01, mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, final))
  acc01 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(acc01, <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, final))
  final &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.selected, cls.noisy02, <span class="dt">seed=</span>j, <span class="dt">classifier=</span>base, <span class="dt">percent=</span><span class="dv">1</span>, <span class="dt">L=</span><span class="dv">10</span>)<span class="op">$</span>final
  ari02 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ari02, mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, final))
  acc02 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(acc02, <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, final))
  final &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.selected, cls.noisy03, <span class="dt">seed=</span>j, <span class="dt">classifier=</span>base, <span class="dt">percent=</span><span class="dv">1</span>, <span class="dt">L=</span><span class="dv">10</span>)<span class="op">$</span>final
  ari03 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ari03, mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, final))
  acc03 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(acc03, <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, final))
  final &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.selected, cls.noisy04, <span class="dt">seed=</span>j, <span class="dt">classifier=</span>base, <span class="dt">percent=</span><span class="dv">1</span>, <span class="dt">L=</span><span class="dv">10</span>)<span class="op">$</span>final
  ari04 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ari04, mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, final))
  acc04 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(acc04, <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, final))
  final &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.selected, cls.noisy05, <span class="dt">seed=</span>j, <span class="dt">classifier=</span>base, <span class="dt">percent=</span><span class="dv">1</span>, <span class="dt">L=</span><span class="dv">10</span>)<span class="op">$</span>final
  ari05 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ari05, mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, final))
  acc05 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(acc05, <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, final))
}

result =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(
  <span class="dt">acc01 =</span> acc01,
  <span class="dt">acc02 =</span> acc02,
  <span class="dt">acc03 =</span> acc03,
  <span class="dt">acc04 =</span> acc04,
  <span class="dt">acc05 =</span> acc05,
  <span class="dt">ari01 =</span> ari01,
  <span class="dt">ari02 =</span> ari02,
  <span class="dt">ari03 =</span> ari03,
  <span class="dt">ari04 =</span> ari04,
  <span class="dt">ari05 =</span> ari05
)

<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/frame">plot.new</a></span>()
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/par">par</a></span>(<span class="dt">mfrow =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(acc01, acc02, acc03, acc04, acc05, <span class="dt">col=</span><span class="st">"lightblue"</span>, <span class="dt">main=</span><span class="st">"SVM Acc"</span>, <span class="dt">ylim=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">0.45</span>, <span class="dv">1</span>))
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">y=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, cls.noisy01), <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, cls.noisy02),
                  <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, cls.noisy03), <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, cls.noisy04),
                  <span class="kw"><a href="reference/bAccuracy.html">bAccuracy</a></span>(lab, cls.noisy05)), <span class="dt">col=</span><span class="st">"red3"</span>, <span class="dt">pch=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>), <span class="dt">cex=</span><span class="dv">1</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(ari01, ari02, ari03, ari04, ari05, <span class="dt">col=</span><span class="st">"lightblue"</span>, <span class="dt">main=</span><span class="st">"SVM ARI"</span>, <span class="dt">ylim=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">0.25</span>, <span class="dv">1</span>))
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">y=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, cls.noisy01), mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, cls.noisy02),
                  mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, cls.noisy03), mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, cls.noisy04),
                  mclust<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/mclust/topics/adjustedRandIndex">adjustedRandIndex</a></span>(lab, cls.noisy05)), <span class="dt">col=</span><span class="st">"red3"</span>, <span class="dt">pch=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>), <span class="dt">cex=</span><span class="dv">1</span>)</code></pre></div>
</div>
<div id="part-b----screclassify-mislabelled-cell-type-correction" class="section level4">
<h4 class="hasAnchor">
<a href="#part-b----screclassify-mislabelled-cell-type-correction" class="anchor"></a>Part B. - scReClassify (mislabelled cell type correction)</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># PCA procedure</span>
dat.pc &lt;-<span class="st"> </span><span class="kw"><a href="reference/matPCs.html">matPCs</a></span>(dat.processed, <span class="fl">0.7</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(dat.pc)

<span class="co"># run scReClassify</span>
cellTypes.reclassify &lt;-<span class="st"> </span><span class="kw"><a href="reference/multiAdaSampling.html">multiAdaSampling</a></span>(dat.pc, cellTypes, <span class="dt">seed =</span> <span class="dv">1</span>, <span class="dt">classifier =</span> <span class="st">"svm"</span>, <span class="dt">percent =</span> <span class="dv">1</span>, <span class="dt">L =</span> <span class="dv">10</span>)

<span class="co"># Verification by marker genes</span>
End &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"KDR"</span>, <span class="st">"LYVE1"</span>)

<span class="co"># check examples</span>
idx &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes.reclassify <span class="op">!=</span><span class="st"> </span>cellTypes)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">cbind</a></span>(<span class="dt">original=</span>cellTypes[idx], <span class="dt">reclassify=</span>cellTypes.reclassify[idx]) <span class="op">%&gt;%</span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>()

c1 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Endothelial Cell"</span>)]
c2 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Erythrocyte"</span>)]
c3 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Hepatoblast"</span>)]
c4 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Macrophage"</span>)]
c5 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Megakaryocyte"</span>)]
c6 &lt;-<span class="st"> </span>dat.processed[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(cellTypes<span class="op">==</span><span class="st">"Mesenchymal Cell"</span>)]
cs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/grDevices/topics/palettes">rainbow</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(cellTypes)))

<span class="co"># (example 1 E13.5_C20)</span>
#####
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/par">par</a></span>(<span class="dt">mfrow=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>))
marker &lt;-<span class="st"> </span>End[<span class="dv">1</span>]
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(c1[marker,], c2[marker,], c3[marker,], c4[marker,], c5[marker,], c6[marker,], <span class="dt">col=</span>cs, <span class="dt">main=</span>marker)
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(<span class="dv">1</span>, dat.processed[marker, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(dat.processed) <span class="op">%in%</span><span class="st"> "E13.5_C20"</span>)], <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">col=</span><span class="st">"red"</span>, <span class="dt">cex=</span><span class="dv">2</span>)
marker &lt;-<span class="st"> </span>End[<span class="dv">2</span>]
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/boxplot">boxplot</a></span>(c1[marker,], c2[marker,], c3[marker,], c4[marker,], c5[marker,], c6[marker,], <span class="dt">col=</span>cs, <span class="dt">main=</span>marker)
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/points">points</a></span>(<span class="dv">1</span>, dat.processed[marker, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(dat.processed) <span class="op">%in%</span><span class="st"> "E13.5_C20"</span>)], <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">col=</span><span class="st">"red"</span>, <span class="dt">cex=</span><span class="dv">2</span>)
#####</code></pre></div>
</div>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ul>
<li>
<strong>scClust</strong>:</li>
</ul>
<p>Kim, T., Chen, I., Lin, Y., Wang, A., Yang, J., &amp; Yang, P.â€  (2018) Impact of similarity metrics on single-cell RNA-seq data clustering. <strong><em>Briefings in Bioinformatics</em></strong> <a href="https://doi.org/10.1093/bib/bby076">[</a><a href="https://doi.org/10.1093/bib/bby076" class="uri">https://doi.org/10.1093/bib/bby076</a>]</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>What license is it under?</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Taiyun Kim <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>

      <footer><div class="copyright">
  <p>Developed by Taiyun Kim.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
